File: Python/programmers_다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    cur_weight = 0
    bridge = []
    
    while len(truck_weights) > 0:
        if len(bridge) == bridge_length:
            cur_weight -= bridge.pop(0)
            
        if cur_weight + truck_weights[0] > weight:
            bridge.append(0)
        else:
            cur = truck_weights.pop(0)
            bridge.append(cur)
            cur_weight += cur
        answer += 1
        
    return answer + bridge_length

File: JS/programmers_다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
    let answer = 0;
    let cur_weight = 0;
    const bridge = [];
    
    while (truck_weights.length) {
        if (bridge.length === bridge_length) {
            cur_weight -= bridge.shift();
        }
        
        if (cur_weight + truck_weights[0] <= weight) {
            cur_weight += truck_weights[0];
            bridge.push(truck_weights.shift())
        } else {
            bridge.push(0);
        }
        
        answer++;
    }
    
    return answer + bridge_length;
}